### DISCUSSION TITLE
PnP Search Filter Web Part V4.9.0 and before -- Multi Value Filter does not work

### DISCUSSION THREAD

**Discussion by Martin-Rosenberg** [microsoft-search/pnp-modern-search#3124] [Ideas]
**Version used**
Ex: 4.9.0 and 4.8.0 

**Describe the bug**

Bug 1A:
The Filter Web Part shows the Option for a multi value managed property - multi choice refiner wrong.

![image](https://github.com/microsoft-search/pnp-modern-search/assets/22766523/37684276-4f54-4dc1-9b13-061d69cdde97)

It should only show us the 3 option "Berlin, London, Paris" and not ervery existing combination. 

Bug 1B:
If you select now Berlin - the filter will not include all items with the value Berlin. Because if the item has e.g. Paris, Berlin it will be not in the result set. 

**To Reproduce**
There is a Site Column from Type Choice Multivalue

![image](https://github.com/microsoft-search/pnp-modern-search/assets/22766523/43d63695-32c3-406c-8acc-33d5c888a5e9)

This Column has a related crawled property: ows_q_CHCM_MultiChoice

This crawled Property is mapped to a managed property from type: RefinableString04

![image](https://github.com/microsoft-search/pnp-modern-search/assets/22766523/4cc30478-eb15-4578-9eb7-672e5e5154db)

In this Column the user can select one or more cities e.g. Paris or Paris, London

The Filter based on this managed property

![image](https://github.com/microsoft-search/pnp-modern-search/assets/22766523/030d0e08-2e9a-4ee7-a4a4-4ea8b26bd933)

the result is not reflecting the configuration
![image](https://github.com/microsoft-search/pnp-modern-search/assets/22766523/1d1e340d-f91a-4076-8f8c-9461a224643e)

(Of course the Search Index was rebuild ...)

**Expected behavior**
I expected to get all items where in the choice field the filtered city is included, 

**Desktop (please complete the following information):**
 - Browser Chrome and Edge (actual version)
 - Tenant at least on 2 different one

**Comment by wobba** [comment]
This is "by design", as the values in the returned properties are concatenated. You should possibly be able to modify the handlebar template for refiners/filters to split on #; and ensure the clicked value works.

Moved this to an idea for improvement to handle concatenated values better.

**Comment by jonathanoh59** [comment]
I am experiencing the same 'problem' and would like to overcome this by modifying the handlebar template. However, when I enter my version of the template, the filter disappear completely.

Has this idea been implemented as an improvement yet? Could an example of how to split on ;# be provided?

Many thanks,
Jonathan
